podpac/core/data/interpolate.py

```diff
@@ -1212,16 +1219,45 @@ def interpolate(self, source_coordinates, source_data, eval_coordinates, output_
         # for debugging purposes, save the last defined interpolator queue
         self._last_interpolator_queue = interpolator_queue
 
+        
+        # With the interpolation queue, intermediate results will not be in the final shape or
+        # coordinate stacking arangement. So, we have to account for this during the loop.
+        intermediate_output = source_data  # Intermediate data that will be interpolated
+        intermediate_coordinates = source_coordinates  # coordinates of intermediate data
+        intermediate_dims_handled = set()  # dimensions that have been handled
+        
         # iterate through each dim tuple in the queue
         for udims in interpolator_queue:
+            if not set(source_coordinates.udims) - intermediate_dims_handled:
+                intermediate_output_data = output_data  # final output, shape of evaluated coordinates
+                output_coordinates = eval_coordinates
+            else:
+                intermediate_dims_handled = intermediate_dims_handled | set(udims)
+                
+                # We need to split the interpolated and yet-to-be-interpolated dimensions
+                # The following dimensions have NOT been interpolated (so output should not
+                # be related to the eval_coordinates)                
+                uneval_dims = [d for d in eval_coordinates.udims if d not in intermediate_dims_handled]
+                # There might be transpose errors here... at least it is a little fragile
+                # This line depends on the order of the dimensions in the interpolator queue
+                output_coordinates = merge_dims(
+                    [eval_coordinates.udrop(uneval_dims), 
+                     intermediate_coordinates.udrop(list(intermediate_dims_handled)),
+                     ] 
+                )
+                # Need intermediate storage -- would be nice if the interpolate call below created this...
+                intermediate_output_data = Node().create_output_array(output_coordinates)
+                
             interpolator = interpolator_queue[udims]
 
             # run interpolation
-            output_data = interpolator.interpolate(udims,
-                                                   source_coordinates,
-                                                   source_data,
-                                                   eval_coordinates,
-                                                   output_data)
-
-        return output_data
-
+            intermediate_output = interpolator.interpolate(udims,
```

Comment:
I had expected the interpolators to use `output_data`  as the `intermediate_output`, similar to how you are using it here. 

Unfortunately the interpolator handling is still pretty fragile (re `loop_helper`...) I think this is part of the iteration for the interpolator where it has first class support for composited data sources in addition to atomic data sources.